src/Workspaces/Core/Portable/PatternMatching/PatternMatcher.cs

```diff
@@ -28,18 +27,15 @@ internal abstract partial class PatternMatcher : IDisposable
         public const int CamelCaseMatchesFromStartBonus = 2;
         public const int CamelCaseMaxWeight = CamelCaseContiguousBonus + CamelCaseMatchesFromStartBonus;
 
-        private readonly object _gate = new object();
-
         private readonly bool _includeMatchedSpans;
         private readonly bool _allowFuzzyMatching;
 
-        private readonly Dictionary<string, StringBreaks> _stringToWordSpans = new Dictionary<string, StringBreaks>();
```

Comment:
No caching.  The saving is minor and it ends up allocating many MBs.  This ends up hurting us as well as it makes us go past our allowed VM buffer.